### PR TITLE
Fix TestILC-related auto-upgrade config

### PR DESCRIPTION
**Pull request by dagood**
See https://github.com/dotnet/corefx/pull/17806

### CODE REVIEW

**Comment by joperezr** [comment]
Merging now to unblock the auto-update PR.

**Comment by dagood** [comment]
Thanks @joperezr.

By the way, tested using `Tools\msbuild.cmd /t:UpdateDependenciesAndSubmitPullRequest /p:ProjectRepoName=corefx;ProjectRepoBranch=master;ProjectRepoOwner=dagood /p:GitHubAuthToken=[...];GitHubUser=dagood-bot;GitHubEmail=dagood-bot@example.com`. I wasn't able to use fake info like I thought in the other thread, since it does some authed `GET`s before doing the upgrades. (I captured the requests with Fiddler and canceled them, instead.) I'll make a buildtools issue to make that experience better.

File: dependencies.props [hunk 1]
```diff
@@ -117,12 +113,12 @@
...
     </XmlUpdateStep>
-    <XmlUpdateStep Condition="'$(DisableProjectNTfsAutoUpgrade)' != 'true'" Include="ProjectNTfs-TestILC">
+    <XmlUpdateStep Condition="'$(DisableProjectNTfsAutoUpgrade)' != 'true'" Include="ProjectNTfsTestILC">
       <Path>$(MSBuildThisFileFullPath)</Path>
       <ElementName>ProjectNTfsTestILCExpectedPrerelease</ElementName>
-      <BuildInfoName>ProjectNTfs-TestILC</BuildInfoName>
+      <BuildInfoName>ProjectNTfsTestILC</BuildInfoName>
     </XmlUpdateStep>
-    <XmlUpdateStep Condition="'$(DisableProjectNTfsAutoUpgrade)' != 'true'" Include="ProjectNTfs-TestILC-PackageVersion">
+    <XmlUpdateStep Condition="'$(DisableProjectNTfsAutoUpgrade)' != 'true'" Include="ProjectNTfsTestILC">
```
Comment (dagood) [inline dependencies.props]: @joperezr For future reference, the itemspec doesn't affect the updater behavior (such as finding the `PackageId`): this upgrader step looks at all BuildInfos. This is why we have `BuildInfoName` metadata on the above updater is there even though it's the same as the spec.

At first having the spec match was just a style convention, but for BfS composed-build auto-upgrade I'm starting to use it to determine which updaters to apply for a composed build.

Comment (joperezr) [inline dependencies.props]: ok cool thanks a lot for the explanation.

File: dependencies.props [hunk 2]
```diff
@@ -117,12 +113,12 @@
       <ElementName>ProjectNTfsExpectedPrerelease</ElementName>
       <BuildInfoName>ProjectNTfs</BuildInfoName>
     </XmlUpdateStep>
-    <XmlUpdateStep Condition="'$(DisableProjectNTfsAutoUpgrade)' != 'true'" Include="ProjectNTfs-TestILC">
+    <XmlUpdateStep Condition="'$(DisableProjectNTfsAutoUpgrade)' != 'true'" Include="ProjectNTfsTestILC">
```
Comment (weshaggard) [inline dependencies.props]: Doesn't these 2 update steps need to have a unique id? Right now it looks like you are only going to get the second one as they both are named ProjectNTfsTestILC and msbuild will simply include again with updated metadata (IIRC).

Comment (dagood) [inline dependencies.props]: I thought the same thing before @ericstj showed me that both items are actually created: https://github.com/dotnet/coreclr/issues/9388#issuecomment-279789940.

**Pull request merged**